jupyter_client/channels.py

```diff
@@ -114,30 +114,18 @@ def _poll(self, start_time: float) -> t.List[t.Any]:
         will be an empty list if no messages arrived before the timeout,
         or the event tuple if there is a message to receive.
         """
+        # Check if an event is waiting
+        events = self.poller.poll(0)
```

Comment:
Does it ensure we poll at least once?

Reply:
If I understand http://api.zeromq.org/2-1:zmq-poll correctly:

> If none of the requested events have occurred on any zmq_pollitem_t item, zmq_poll() shall wait timeout microseconds for an event to occur on any of the requested items. If the value of timeout is 0, zmq_poll() shall return immediately. If the value of timeout is -1, zmq_poll() shall block indefinitely until a requested event has occurred on at least one zmq_pollitem_t. The resolution of timeout is 1 millisecond.

A timeout of 0 means that if a message is already there it will be polled.

Reply:
> A timeout of 0 means that if a message is already there it will be polled.

Thanks, although [pyzmq doesn't mention that](https://pyzmq.readthedocs.io/en/latest/api/zmq.html?highlight=poll#zmq.Poller.poll).